R/state.R

```diff
@@ -0,0 +1,99 @@
+
+# TODO: Use our own stack of expressions
+scoped_exit <- function(expr, frame = caller_env()) {
+  if (is_reference(frame, global_env())) {
+    abort("Can't add an exit event at top-level")
+  }
+
+  expr <- enexpr(expr)
+
+  # Inline everything so the call will succeed in any environment
+  expr <- lang(on.exit, expr, add = TRUE)
+  eval_bare(expr, frame)
+
+  invisible(expr)
+}
+
+#' Change global options
+#'
+#' @description
+#'
+#' * `scoped_options()` changes options for the duration of a stack
```

Comment:
Seeing these all together makes me think that the "option(s)" should be a prefix, not a suffix?

Reply:
I think the prefix should be `scoped_` because we've already started using it, e.g. `scoped_vars()`. Also this makes these families more consistent: `with_`, `scoped_`, `poke_`, `peek_`.